### PR TITLE
feat(models): add BirdNET v3.0 acoustic classifier support

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Adds a working, tested backend inference path for the BirdNET v3.0 acoustic classifier (EfficientNetV2-S backbone, 11,560 species, 32 kHz / 5 s window), gated so BirdNET v2.4 stays the default. v3.0 is a preview-track effort: it reuses the existing 32 kHz / 5 s buffer path (shared with Perch v2), so the audio pipeline needs no new buffer, and the catalog entry is kept hidden until the model repository goes public at GA.

Key points:

- New `BirdNETV3` model instance mirroring the Perch v2 path: OpenVINO-first with ONNX Runtime fallback, but with no model-variant filename gate because the GPU-native v3.0 graph (Conv1d front-end, no `STFT` op) runs on OpenVINO directly.
- v3.0's `predictions` output is a per-class sigmoid applied in-graph, so it is used verbatim with no re-activation (unlike Perch's softmax and v2.4's sigmoid). This also fixes a latent double-sigmoid in the shared ONNX post-processing for the v3.0 model type (reachable only via the high-level Predict path, not the production instance path).
- The embeddings and predictions output ports are resolved by size, not fixed position, so both ONNX export orderings load correctly.
- `Scientific name_Common name` label parsing (identical format to BirdNET v2.4), so common names resolve directly with no extra name resolver.
- New `conf.BirdNETV3Config` plus settings / orchestrator loader / model-manager wiring; the v3 geomodel range filter already auto-selects for v3.0.

Not in this PR (preview-track follow-ups): the model repository is private during preview, so the gallery download entry stays hidden and its checksums land later; the regional model catalog and the settings model-selection UI are separate follow-ups.

## Validation

- New unit tests: label parsing, the no-activation output contract, size-based port detection (both export orderings), orchestrator/loader registration, catalog entry shape, and the Predict error branches.
- End-to-end real-inference check through the ONNX Runtime path on real 32 kHz audio: a tawny owl clip yields `Strix aluco` (Tawny Owl) at 0.92 confidence with every other species near zero; a multi-species soundscape yields a realistic probability spread (no double-sigmoid flattening).
- `golangci-lint` clean, `go test -race` green for the affected packages, `config.schema.json` and the configuration reference regenerated, and the hot-reload coverage and schema-drift guards pass.

## Test Plan

- [ ] CI green (build, unit tests, lint, cross-platform)
- [ ] Automated review addressed

## Summary by CodeRabbit

* **New Features**
  * Added BirdNET v3.0 acoustic classification support with ONNX Runtime and optional OpenVINO acceleration.
  * Added configurable model and label paths, confidence threshold, and locale settings.
  * Added automatic model installation, loading, prediction, and runtime detection.
  * Updated the model catalog with BirdNET v3.0 metadata and 11,560 species labels.
* **Documentation**
  * Added BirdNET v3.0 settings to the configuration reference.
* **Bug Fixes**
  * Improved model output detection across different export formats.
  * Ensured model settings are correctly saved during installation and cleared during removal.

### CODE REVIEW

File: internal/conf/config.go [hunk 1]
```diff
@@ -1730,11 +1738,12 @@ type Settings struct {
...
 
-	BirdNET BirdNETConfig `yaml:"birdnet" json:"birdnet"` // BirdNET configuration
-	Perch   PerchConfig   `yaml:"perch" json:"perch"`     // Perch v2 model configuration
-	Bat     BatConfig     `yaml:"bat" json:"bat"`         // Bat detection configuration
-	BSG     BSGConfig     `yaml:"bsg" json:"bsg"`         // BSG regional bird model configuration
-	Models  ModelsConfig  `yaml:"models" json:"models"`   // Global model enablement and management
+	BirdNET   BirdNETConfig   `yaml:"birdnet" json:"birdnet"`     // BirdNET configuration
+	Perch     PerchConfig     `yaml:"perch" json:"perch"`         // Perch v2 model configuration
+	BirdNETV3 BirdNETV3Config `yaml:"birdnetv3" json:"birdNetV3"` // BirdNET v3.0 acoustic classifier configuration
+	Bat       BatConfig       `yaml:"bat" json:"bat"`             // Bat detection configuration
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/conf/config.go:1744]: Good catch, fixed in 382557120. Changed the tag to `json:"birdnetv3"` so it matches the YAML key and the sibling perch/bat/bsg sections, and the full-settings response now lines up with the generated schema and docs.

File: internal/inference/onnx/postprocess.go [hunk 2]
```diff
@@ -39,6 +39,27 @@ func softmax(logits []float32) []float32 {
...
+		return sigmoidSlice(logits)
+	case BirdNETv30:
+		return slices.Clone(logits)
+	case PerchV2:
+		return softmax(logits)
+	}
+	// Defensive fallback for an unhandled model type. ModelType is a closed
+	// three-value enum, all handled above, so this is unreachable in practice.
+	return sigmoidSlice(logits)
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/inference/onnx/postprocess.go:64]: Fixed in 382557120. `softmax` now returns a fresh empty slice for empty input instead of handing back the argument, so the "newly allocated, never aliases logits" contract holds on every path.

File: internal/api/v2/hotreload_coverage_test.go [hunk 3]
```diff
@@ -71,6 +71,9 @@ var hotReloadRegistry = map[string]hotReloadEntry{
 	// --- Perch ---
 	"Perch": {categories: []hotReloadCategory{hotReloadRestart}},
 
+	// --- BirdNET v3.0 ---
+	"BirdNETV3": {categories: []hotReloadCategory{hotReloadRestart}},
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/api/v2/hotreload_coverage_test.go:75]: Fair observation, but I'm leaving this as restart for now and tracking it separately.

BirdNET v3.0 deliberately mirrors the existing secondary-model convention: `Perch` and `BSG` are both registered as `hotReloadRestart` with the same "not wired" exemption, because none of the secondary model paths (`perch.modelpath`, `bsg.modelpath`, and now `birdnetv3.modelpath`) have a settings-change detector wired into `restartRequiringChecks`. The orchestrator does support rebuilding these instances, but the detector side is the missing piece, and it's missing for all three equally.

Wiring it up for v3.0 alone would make it the odd one out among its siblings and is a bigger change than this PR should carry. v3.0 is also preview-gated (the catalog entry is hidden and the model repo is private until GA), so nobody is editing these paths from the UI yet. I'd rather do the secondary-model reload path once, for all three, in a follow-up.

[the remark this answers is not in the document]
Comment (tphakala) [reply internal/api/v2/hotreload_coverage_test.go:75]: Thanks. No need for a GitHub issue, I'm tracking the shared secondary-model reload wiring internally.
